### PR TITLE
Implement GetHitTestResults for ARKit

### PR DESCRIPTION
**Pull request by Alex-MSFT**
This change implements the iOS ARKit polyfill for GetHitTestResults as detailed in #278 .  This is the ARKit equivalent of #240.  

This supports hit testing via ARRaycastQuery for iOS 13+ and ARFrame.hitTest for iOS <=12.0.  For versions less than 13.0 only hit testing from the center of the screen is supported.  We will need to come up with a scheme for translating a world space ray to a screen point, as casting from a world space ray is not supported in older versions.  Since we currently don't support setting an offset ray at the API level I will defer this for now.

### CODE REVIEW

File: Dependencies/xr/Source/ARKit/XR.mm [hunk 1]
```diff
@@ -61,12 +61,11 @@ - (UIInterfaceOrientation)orientation {
...
     if (@available(iOS 13.0, *)) {
-        if (window && window.windowScene) {
-            return window.windowScene.interfaceOrientation;
-        }
+        return window.windowScene.interfaceOrientation;
     }
-
-    return [[UIApplication sharedApplication] statusBarOrientation];
+    else {
+        return [[UIApplication sharedApplication] statusBarOrientation];
```
Comment (ryantrem) [inline Dependencies/xr/Source/ARKit/XR.mm:67]: I don't think you need this change, but either way.

File: Dependencies/xr/Source/ARKit/XR.mm [hunk 2]
```diff
@@ -574,10 +571,52 @@ void DrawFrame() {
...
+                                             alignment:ARRaycastTargetAlignmentAny];
+                        
+                        // Perform the actual raycast.
+                        auto rayCastResults = [session raycast:raycastQuery];
+                        
+                        // Process the results and push them into the results list.
+                        for (ARRaycastResult* result in rayCastResults) {
+                            filteredResults.push_back(transformToHitResult(result.worldTransform));
+                        }
+                    } else {
```
Comment (ryantrem) [inline Dependencies/xr/Source/ARKit/XR.mm:611]: If `offsetRay`'s origin is where the camera is (raycasting from the camera), then couldn't we hit test to a correct (according to the Ray) screen coordinate?

File: Dependencies/xr/Source/ARKit/XR.mm [hunk 3]
```diff
@@ -574,10 +571,52 @@ void DrawFrame() {
...
+                        auto hitTestResults = [session.currentFrame hitTest:CGPointMake(.5, .5) types:(ARHitTestResultTypeExistingPlane)];
+                        for (ARHitTestResult* result in hitTestResults) {
+                            filteredResults.push_back(transformToHitResult(result.worldTransform));
+                        }
+                    }
+                }
+            }
         }
-        
+
```
Comment (ryantrem) [inline Dependencies/xr/Source/ARKit/XR.mm:621]:
```suggestion

```

File: Dependencies/xr/Source/ARKit/XR.mm [hunk 4]
```diff
@@ -589,6 +628,23 @@ void GetHitTestResults(std::vector<HitResult>&, xr::Ray) const {
...
+        /*
+         Helper function to translate a world transform into a hit test result.
+         */
+        HitResult transformToHitResult(simd_float4x4 transform) const {
+            auto orientation = simd_quaternion(transform);
+            HitResult hitResult{};
+            hitResult.Pose.Orientation = { orientation.vector.x,
+                orientation.vector.y,
+                orientation.vector.z,
+                orientation.vector.w};
```
Comment (ryantrem) [inline Dependencies/xr/Source/ARKit/XR.mm:643]:
```suggestion
            hitResult.Pose.Orientation = {
                orientation.vector.x,
                orientation.vector.y,
                orientation.vector.z,
                orientation.vector.w
            };
```

File: Dependencies/xr/Source/ARKit/XR.mm [hunk 5]
```diff
@@ -589,6 +628,23 @@ void GetHitTestResults(std::vector<HitResult>&, xr::Ray) const {
...
+            hitResult.Pose.Position = { transform.columns[3][0],
+                transform.columns[3][1],
+                transform.columns[3][2] };
```
Comment (ryantrem) [inline Dependencies/xr/Source/ARKit/XR.mm:646]:
```suggestion
            hitResult.Pose.Position = {
                transform.columns[3][0],
                transform.columns[3][1],
                transform.columns[3][2]
            };
```

File: Dependencies/xr/Source/ARKit/XR.mm (hunk 2 above)
Comment (ryantrem) [reply Dependencies/xr/Source/ARKit/XR.mm:611]: Oh I see now you commented in the PR saying we could do this later.

Comment (Alex-MSFT) [reply Dependencies/xr/Source/ARKit/XR.mm:611]: Yeah I'll do this later, currently we don't actually pass the offset ray in so effectively we're always hit testing from the center point

**Pull request merged**